File: Euclid/Messages/Outgoing/ORDERINFO.cs
using Euclid.Network.Streams.Util;

namespace Euclid.Messages.Outgoing
{
    class ORDERINFO : IMessageComposer
    {
        private string id;
        private int price;
        private string customData;
        private string name;

        public ORDERINFO(string id, int price, string customData, string name)
        {
            this.id = id;
            this.price = price;
            this.customData = customData;
            this.name = name;
        }

        public override void Write()
        {
            this.Data.Add(new ArgumentEntry(id));
            this.Data.Add(new ArgumentEntry(price));
            this.Data.Add(new ArgumentEntry(customData));
            this.Data.Add(new ArgumentEntry(name));
        }
    }
}

File: euclid/src/main/java/euclid/messages/outgoing/ORDERINFO.java
package euclid.messages.outgoing;

import euclid.messages.IMessageComposer;
import euclid.network.streams.util.ArgumentEntry;

public class ORDERINFO extends IMessageComposer {

    private final String id;
    private final int price;
    private final String customData;
    private final String name;

    public ORDERINFO(String id, int price, String customData, String name) {
        this.id = id;
        this.price = price;
        this.customData = customData;
        this.name = name;
    }

    @Override
    public void write() {
        getData().add(new ArgumentEntry(id));
        getData().add(new ArgumentEntry(price));
        getData().add(new ArgumentEntry(customData));
        getData().add(new ArgumentEntry(name));
    }
}
